sdk/search/azure-search-documents/azure/search/documents/_index/_search_index_client.py

```diff
@@ -110,6 +102,7 @@ def get_document_count(self, **kwargs):
 
         :rtype: int
         """
+        kwargs.setdefault("headers", self._headers)
```

Comment:
Not sure what is best here tbh. Is it expected/common that users will want to pass their own `headers` param? If so, should this merge rather than overwrite?

Reply:
Well, I've gone ahead and updated to merge with any passed-in headers